agrirouter-api-java-convenience/src/main/java/com/dke/data/agrirouter/convenience/rest/service/messaging/MessageConfirmationFunctionsService.java

```diff
@@ -101,7 +103,8 @@ private void validateResponse(MessageConfirmationForAllPendingMessagesParameters
     DecodeMessageResponse decodedMessageQueryResponse;
     fetchMessageResponses =
         this.fetchMessageService.fetch(
-            parameters.getOnboardingResponse(), MAX_TRIES_BEFORE_FAILURE, DEFAULT_INTERVAL);
+            parameters.getOnboardingResponse(),
+            new DefaultCancellationToken(MAX_TRIES_BEFORE_FAILURE, DEFAULT_INTERVAL));
```

Comment:
Fix:

`    fetchMessageResponses =
        this.fetchMessageService.fetch(
            parameters.getOnboardingResponse(),
            this.cancellationToken);`

Reply:
Adding the cancellation token to the constructor would mean, that you need to modify for example DI for frameworks like Spring Boot and provide a dedicated bean. This would have a big impact at this point. Since we are talking about a convenience implementation I would suggest using the default behavior.

Reply:
OK, agree. Maybe add a setCancelationToken() function instead? 

public void setCancellationToken(CancellationToken cancellationToken){
   this.cancellationToken = cancellationToken;
}

Would mean you have to check this.cancellationToken before calling it. If it's empty, create a DefaultCancellationToken. Or create a DefaultCancellationToken and overwrite it by the setter.

Reply:
I will find a solution for it.